project-base/storefront/cypress/e2e/authentication/registration.cy.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    goToRegistrationPageFromHeader,
+    fillInRegstrationForm,
+    checkRegistrationValidationErrorsPopup,
+} from './authenticationSupport';
+import { DEFAULT_APP_STORE, url } from 'fixtures/demodata';
+import { generateCustomerRegistrationData } from 'fixtures/generators';
+import {
+    checkAndHideErrorToast,
+    checkAndHideSuccessToast,
+    checkUrl,
+    goToEditProfileFromHeader,
+    takeSnapshotAndCompare,
+} from 'support';
+import { TIDs } from 'tids';
+
+describe('Registration tests', () => {
+    beforeEach(() => {
+        cy.window().then((win) => {
+            win.localStorage.setItem('app-store', JSON.stringify(DEFAULT_APP_STORE));
+        });
+    });
+
+    it('should register as a B2C customer', () => {
+        cy.visit('/');
+
+        goToRegistrationPageFromHeader();
+        fillInRegstrationForm('commonCustomer', 'registraion-as-b2c@shopsys.com');
+
+        takeSnapshotAndCompare('b2c-registration-page');
+
+        cy.getByTID([TIDs.registration_submit_button]).click();
```

Comment:
This is a result of quite extensive discussion among FE developers. Initially, we had data-testId, which we could pass to HTML, but could not use in the code (react does not understand kebab-case). Because of that, we were looking for a more suitable name, and agreed on `tid`. I understand that it might not be completely to your liking, but as I say, there was already a lot of discussions, we agreed on it, projects already applied those changes, and changing it again would be a pain in the ass, I guess.